src/libs/OptionsListUtils.js

```diff
@@ -603,6 +603,12 @@ function getOptions(
         };
     }
 
+    // We're only picking personal details that have logins set
+    // This is a temporary fix for all the logic that's been breaking because of the new privacy changes
+    // See https://github.com/Expensify/Expensify/issues/293465 for more context
+    // eslint-disable-next-line no-param-reassign
+    personalDetails = _.pick(personalDetails, (detail) => Boolean(detail.login));
```

Comment:
What are the effects of excluding personal details from this object? They don't appear in the option list?

Do you know where this was causing problems down the line? I imagine we where trying to use `personalDetails.login` which was missing, but in other fixes we have been adding `personalDetails.login || personalDetails.displayName`... just wondering why that approach was not good here.

Reply:
Because Tasks takes the login from the option and does a lot more stuff with it, so we can't just do that

Also, it's not guaranteed that personalDetails.displayName is equal to their login. If they have a display name set already then it won't be equal.

Reply:
I assume that we actually needed the login (emails) for the flows that broke downstream, so using the `displayName` might not be enough

Reply:
> That only applies to new users

Then it can be that the `personalDetails.login` is missing for users that are not new? I mean, can it happen that `personalDetails.login` is missing and `personalDetails.displayName` is missing?

Reply:
>In the cases I have seen so far, when the login is missing, the displayName is the email address.

Is this still true if the user sets their display name?

Reply:
If you start a DM with someone who isn't on the same workspace/domain as you, then personalDetails.login will be missing. If that individual has a display name set, then their login won't be stored in their displayName.

Reply:
> > In the cases I have seen so far, when the login is missing, the displayName is the email address.
> 
> Is this still true if the user sets their display name?

I don't know really, I always thought that the accounts missing the `login` where accounts that you added somewhere by email, but still have not fetched the personal details from the backend.. or something like that.

The point is: if an account without `login` can have a `displayName` that is not an email, then maybe the fixed `login || displayName` may be wrong?

Reply:
@aldo-expensify Can you test your PR with this?

> If you start a DM with someone who isn't on the same workspace/domain as you, then personalDetails.login will be missing. This is expected behavior for privacy concerns. If that individual has a display name set, then their login won't be stored in their displayName.

Reply:
Yep, that's what I'm going to do.

How do you start a DM with someone without using the email? do you have to find this account in a public room?

Reply:
I did this:

- Account A creates a workspace with a public room
- Create account B on a different domain and sets a display name
- With account B, open the public room using the report ID (success)
- Then what?

I tried commenting with Account A, then account B creates a thread, but at that point if Account A tries to see the Account B personal details, it is broken (endless spinner):

<img width="805" alt="image" src="https://github.com/Expensify/App/assets/87341702/e601eb71-4dc5-4c6d-8423-7ef436cebfcd">

Reply:
> If you start a DM with someone who isn't on the same workspace/domain as you, then personalDetails.login will be missing.

@jasperhuangg do you have steps to do this? how do you start such DM?

Reply:
Hmm, I don't think there's a way to start a DM with someone that you don't have a DM with if we do the changes in this PR. 

If we don't do the changes here, then if you are in a public room, you will see the people that have posted in that room in the search and will have the option to create a DM with them.

Reply:
Looks like this has changed the expected behavior with 1:1 chats. Like if I have 1:1 chat with some user and type their email in chat switcher. It doesn't show up and pretends the user doesn't exist and creates a new chat instead

Reply:
@MonilBhavsar aah that may be because if you send someone a 1:1 DM, you don't get their personal details stored in your Onyx, because that could be a way to DM a bunch of people and see if their emails exist on NewDot, right?

Reply:
Hmm, it should return email if another person also replies?

Reply:
yes i think it should - if it doesn't now we should make it happen haha

The idea is: If I send YOU a message, I let YOU know my contact info (and vise versa)

Reply:
Yes, makes sense. Looks like it is not working like this right now. Like, some active 1:1 DM's are not displayed in chat switcher

Reply:
Ok I think we can open up an issue about this - it should be a quick fix, I'd think?

For group DMs I think you'd also send your personal details to everyone in the room, right?

And for domain / workspace rooms everyone should have your details already, and public rooms we DON'T want to send your details, so only 1:1 or group DMS

Reply:
I guess this is because of @puneetlath's query was reverted [here](https://github.com/Expensify/Auth/pull/8142/files#diff-55e1cdd661dc1366de9fba4518bc232a4a1d7d28e39c3ac1388fab0d47dd7ad0R3248) and we're no longer marking user's in 1:1 or group chats known?

Reply:
ok, [this](https://github.com/Expensify/Auth/pull/8153) Auth fix works for 1:1 DM's

Reply:
It's intended to also work for group DMs

Reply:
Now to the last part of the issue, where if I have DM some random new users and send some messages. The chat appears in LHN. Again, when I go to chat switcher and type full email, and click on option, it creates a new report with same user

Reply:
I reported [here](https://expensify.slack.com/archives/C049HHMV9SM/p1687354648486659), in case it is already known